gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -950,6 +950,7 @@ private JobStatus pollStatus(String flowGroup, String flowName, long flowExecuti
     synchronized Map<String, Set<DagNode<JobExecutionPlan>>> submitNext(String dagId) throws IOException {
       Dag<JobExecutionPlan> dag = this.dags.get(dagId);
       Set<DagNode<JobExecutionPlan>> nextNodes = DagManagerUtils.getNext(dag);
+      log.info("Submitting next nodes for dagId {}, where next nodes are {}", dagId, nextNodes);
```

Comment:
Are we sure that nextNodes is safe to log and won't just be an object reference? I think you may need to get a list of jobNames within nextNodes by iterating through it.

Reply:
extracting the names and printing to be safe